c7n/resources/ec2.py

```diff
@@ -295,6 +297,68 @@ def __call__(self, i):
         return self.operator(map(self.match, volumes))
 
 
+@filters.register('stop-protected')
+class DisableApiStop(Filter):
+    """EC2 instances with ``disableApiStop`` attribute set
+
+    Filters EC2 instances with ``disableApiStop`` attribute set to true.
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: stop-protection-enabled
+            resource: ec2
+            filters:
+              - type: stop-protected
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: stop-protection-NOT-enabled
+            resource: ec2
+            filters:
+              - not:
+                - type: stop-protected
+    """
+
+    schema = type_schema('stop-protected')
+    permissions = ('ec2:DescribeInstanceAttribute',)
+
+    def get_permissions(self) -> List[str]:
+        perms = list(self.permissions)
+        perms.extend(self.manager.get_permissions())
```

Comment:
you can omit this method, manager perms are always part of the aggregated permission set for a policy, and the class attribute for the permission suffices for the base class implementation of the method.

Reply:
Thanks for information. Deleted.